packages/yarnpkg-core/sources/Cache.ts

```diff
@@ -221,13 +221,19 @@ export class Cache {
       if (!cachePath)
         throw new Error(`Assertion failed: Expected the cache path to be available`);
 
+      const potentialMirrorPath =
+        packageSource === `mirror` &&
+        (expectedChecksum === null || expectedChecksum === checksum)
```

Comment:
Is there a particular reason for the checksum check here?

Reply:
Preparing for `YARN_CHECKSUM_BEHAVIOR=reset` I suppose, but other than that It's fine to remove it 🤔

Reply:
Removed